extension/services/applemusic/player.js

```diff
@@ -0,0 +1,61 @@
+/* globals helpers */
+
+this.player = (function() {
+  const SEARCH_PLAY = ".lockup__controls .play-button";
+
+  class Player extends helpers.Runner {
+    action_play() {
+      const button = this.querySelector(SEARCH_PLAY);
+      button.click();
+    }
+
+    async action_search({ query, thenPlay }) {
+      const searchInput = await this.waitForSelector(
+        "#search-box .dt-search-box__input"
+      );
+      const pressEnter = new KeyboardEvent("keydown", {
+        bubbles: true,
+        cancelable: true,
+        keyCode: 13,
+      });
+      searchInput.onfocus = function() {
+        document.body.dispatchEvent(pressEnter);
```

Comment:
Generally you would do `searchInput.dispatchEvent(pressEnter)`

Though in my testing that also doesn't work. If I do a "keyup" enter event then it shows the dropdown, but doesn't search:

<img width="234" alt="image" src="https://user-images.githubusercontent.com/44390/77368887-3c3eee80-6d2b-11ea-80ab-dcd44a87c6d2.png">

But I think the one thing that is missing is `key: "Enter"` instead of looking at `keyCode`. E.g., if I do `searchInput.dispatchEvent(new KeyboardEvent("keyup", {key: "Enter"}))` it works

Reply:
Oh! thanks @ianb